File: src/main/java/org/hisp/dhis/rules/models/RuleActionCreateEvent.java
package org.hisp.dhis.rules.models;

import javax.annotation.Nonnull;
import javax.annotation.CheckForNull;

public record RuleActionCreateEvent(
        @Nonnull String data,
        @Nonnull String content,
        @Nonnull String programStage
) implements RuleAction {

    @Nonnull
    public static RuleActionCreateEvent create(@CheckForNull String content,
                                               @CheckForNull String data, @Nonnull String programStage) {
        return new RuleActionCreateEvent(data == null ? "" : data, content == null ? "" : content, programStage);
    }
}

File: src/main/java/org/hisp/dhis/rules/models/RuleActionCreateEvent.kt
package org.hisp.dhis.rules.models


data class RuleActionCreateEvent(val content: String?,
                                 val data: String?,
                                 val programStage: String?) : RuleAction() {

    companion object {

        @JvmStatic
        fun create(content: String?, data: String?, programStage: String) =
                RuleActionCreateEvent(content ?: "", data ?: "", programStage)
    }
}